automates/text_reading/src/main/scala/org/clulab/grounding/WikiGrounder.scala

```diff
@@ -0,0 +1,168 @@
+          lineResults += lineResult
+
+        }
+
+        // when there are multiple subclassOf per wikidata entry, each returned separately, so we need to make one new wiki result object with all of the subclass entries
+        val grouped = lineResults.groupBy(_.conceptID)
+        val newLineResults = new ArrayBuffer[sparqlWikiResult]()
+        for (g <- grouped) {
+          if (g._2.length > 1) {
+
+            val allClassOf = Some(g._2.map(_.subClassOf.getOrElse("NA")).mkString(","))
+            val oneResultInGroup = g._2.head
+            newLineResults.append(new sparqlWikiResult(oneResultInGroup.searchTerm, oneResultInGroup.conceptID, oneResultInGroup.conceptLabel, oneResultInGroup.conceptDescription, oneResultInGroup.alternativeLabel, allClassOf, oneResultInGroup.score, oneResultInGroup.source))
+          } else newLineResults.append(g._2.head)
+        }
+
+
+        val allLabels = newLineResults.map(res => res.conceptLabel)
+        val duplicates = allLabels.groupBy(identity).collect { case (x, ys) if ys.lengthCompare(1) > 0 => x }.toList
+        val (uniqueLabelResLines, nonUniqLabelResLines) = newLineResults.partition(res => !duplicates.contains(res.conceptLabel))
+        allSparqlWikiResults ++= uniqueLabelResLines
+        //out of the items with the same label, e.g., crop (grown and harvested plant or animal product) vs. crop (hairstyle), choose the one with the highest score based on similarity of the wikidata description and alternative label to the sentence and search term the search term from
+        if (nonUniqLabelResLines.nonEmpty) {
+          allSparqlWikiResults += nonUniqLabelResLines.maxBy(_.score)
+        }
+
+      }
+      resultsFromAllTerms ++= allSparqlWikiResults
+      // update `cache` file
+      exporter.export(ujson.write(fileCache), cacheFilePath.replace(".json", ""))
+    }
+
+    Some(getTopKWiki(resultsFromAllTerms.toList.distinct.sortBy(_.score).reverse, k))
+
+  } else None
+
+}
+
+  def getTopK(results: Seq[sparqlResult], k: Int): Seq[sparqlResult] = {
+    if (k < results.length) {
+      results.slice(0,k)
+    } else {
+      results
+    }
+  }
+
+  def getTopKWiki(results: Seq[sparqlWikiResult], k: Int): Seq[sparqlWikiResult] = {
+    if (k < results.length) {
+      results.slice(0,k)
+    } else {
+      results
+    }
+  }
+
+  def runSparqlQuery(term: String, scriptDir: String): String = {
+    val command = Seq("python", s"$sparqlDir/sparqlWikiWrapper.py", term)
+    val process = Process(command, new File(s"$scriptDir"))
+    process.!!
+  }
+
+  def editDistance(s1: String, s2: String): Double = {
+    val dist = LevenshteinDistance.getDefaultInstance().apply(s1, s2).toDouble
+    dist
+  }
+
+  def editDistanceNormalized(s1: String, s2: String): Double = {
+    val maxLength = math.max(s1.length, s2.length)
+    val levenshteinDist = LevenshteinDistance.getDefaultInstance().apply(s1, s2).toDouble
```

Comment:
@maxaalexeeva replace this with a call to the function `editDistance()` defined above. It is common practice to reuse functions even for single lines of code. In the future, we may wish to edit the `LevenshteinDistance` call or extend it with additional steps. It's best to only have to make those changes in a single location instead of trying to keep two separate but exactly the same pieces of code in sync with each other!

Reply:
good eye! thanks!